src/components/clusterConfiguration/ClusterConfiguration.tsx

```diff
@@ -1,332 +1,38 @@
-import alertsReducer, {
-  addAlert,
-  AlertProps,
-  removeAlert,
-} from '../../features/alerts/alertsSlice';
-import BaremetalInventory from './BaremetalInventory';
-import {
-  nameValidationSchema,
-  sshPublicKeyValidationSchema,
-  validJSONSchema,
-  ipBlockValidationSchema,
-  dnsNameValidationSchema,
-  hostPrefixValidationSchema,
-  vipValidationSchema,
-} from '../ui/formik/validationSchemas';
-import ClusterBreadcrumbs from '../clusters/ClusterBreadcrumbs';
-import { HostSubnets, ClusterConfigurationValues } from '../../types/clusters';
-import NetworkConfiguration from './NetworkConfiguration';
-import ClusterValidationSection from './ClusterValidationSection';
-import { validateCluster } from './clusterValidations';
-import { getInitialValues, getHostSubnets, findMatchingSubnet } from './utils';
-
-const validationSchema = (hostSubnets: HostSubnets) =>
-  Yup.lazy<ClusterConfigurationValues>((values) =>
-    Yup.object<ClusterConfigurationValues>().shape({
-      name: nameValidationSchema,
-      baseDnsDomain: dnsNameValidationSchema,
-      clusterNetworkHostPrefix: hostPrefixValidationSchema,
-      clusterNetworkCidr: ipBlockValidationSchema,
-      serviceNetworkCidr: ipBlockValidationSchema,
-      apiVip: vipValidationSchema(hostSubnets, values),
-      ingressVip: vipValidationSchema(hostSubnets, values),
-      pullSecret: validJSONSchema,
-      sshPublicKey: sshPublicKeyValidationSchema,
-    }),
-  );
-
-const sshPublicKeyHelperText = (
-  <>
-    SSH public key for debugging OpenShift nodes, value of <em>~/.ssh/id_rsa.pub</em> can be
-    copy&amp;pasted here. To generate new pair, use <em>ssh-keygen -o</em>.
-  </>
-);
 
 type ClusterConfigurationProps = {
   cluster: Cluster;
 };
-
 const ClusterConfiguration: React.FC<ClusterConfigurationProps> = ({ cluster }) => {
-  const [isValidationSectionOpen, setIsValidationSectionOpen] = React.useState(false);
-  const [isStartingInstallation, setIsStartingInstallation] = React.useState(false);
-  const dispatch = useDispatch();
-  const [alerts, dispatchAlertsAction] = React.useReducer(alertsReducer, []);
-  const hostSubnets = getHostSubnets(cluster);
-
-  const handleSubmit = async (
-    values: ClusterConfigurationValues,
-    formikActions: FormikHelpers<ClusterConfigurationValues>,
-  ) => {
-    // async validation for cluster name - run only on submit
-    try {
-      const { data: clusters } = await getClusters();
-      const names = clusters.map((c) => c.name).filter((n) => n !== cluster.name);
-      if (names.includes(values.name)) {
-        return formikActions.setFieldError('name', `Name "${values.name}" is already taken.`);
+  const [domains, setDomains] = React.useState<ManagedDomain[] | undefined>();
+
+  React.useEffect(() => {
+    const fetchManagedDomains = async () => {
+      try {
+        const { data } = await getManagedDomains();
+        setDomains(data);
+      } catch (e) {
+        setDomains([]);
+        handleApiError(e);
```

Comment:
Shouldnt we also notify user that we could not fetch the domains via alert `dispatchAlertsAction` ?

Reply:
The list of domains is not something that user can influence, so I opted to fail silently (by logging the error in the console) and fall back to empty list of domains. If the list of domains is empty user won't see the option to select to use the RH provided domains.

Reply:
Its true that user cannot influence it, but he still may be left wondering what happened to managed domains ? I still prefer transparency over failing silently. I think any request that fails and shouldn't  (meaning its an actual error) should be visualized to user. But thats just my preference, if you think the current solution is better then lets go with it.

Reply:
OK, I've enhanced the error handling to include errors for fetching the domains.